core/spring-boot/src/main/java/org/springframework/boot/logging/log4j2/Log4J2LoggingSystem.java

```diff
@@ -123,18 +119,18 @@ public Log4J2LoggingSystem(ClassLoader classLoader) {
 	protected String[] getStandardConfigLocations() {
 		List<String> locations = new ArrayList<>();
 		locations.add("log4j2-test.properties");
-		if (isClassAvailable("com.fasterxml.jackson.dataformat.yaml.YAMLParser")) {
```

Comment:
What about Jackson 3? Does log4J2 supports it or does it need to rely on Jackson 2 only still?

Reply:
We use both Jackson 2 and 3, but only the presence of Jackson 2 needs to be tested for:

- `log4j-core` version 2.x has an optional dependency on `jackson-dataformat-yaml`, which activates the embedded `ConfigurationFactory`.
- In Log4j 3, the configuration factory has been moved to `log4j-config-yaml`, which has a **non** optional dependency on `jackson-dataformat-yaml` (currently version 2, but will use version 3 after apache/logging-log4j2#3701) has been merged. Since the dependency is not optional, there is no need to test for it.